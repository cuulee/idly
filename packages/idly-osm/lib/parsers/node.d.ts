import { Node, NodeGeometry, ParentWays, OsmGeometry } from 'idly-common/lib/osm/structures';
export declare const DEFAULT_NODE_ICON = "circle";
export declare function nodePropertiesGen(node: Node, parentWays: ParentWays): {
    icon: any;
    name: string;
    geometry: OsmGeometry;
};
export declare const applyNodeMarkup: (geometry: NodeGeometry, tags: Map<string, string>) => {
    icon: any;
    name: string;
    geometry: OsmGeometry;
};
