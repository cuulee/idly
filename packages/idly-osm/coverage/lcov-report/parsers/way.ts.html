<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for parsers/way.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">parsers</a> way.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { weakCache } from 'idly-common/lib/misc/weakCache';
import {
  EntityTable,
  OsmGeometry,
  Way,
  EntityType
} from 'idly-common/lib/osm/structures';
<span class="branch-0 cbranch-no" title="branch not covered" ></span>
import { isArea } from '../helpers/isArea';
import { tagClassesPrimary } from '../tagClasses/tagClasses';
import { presetsMatcherCached } from '../presets/presets';
&nbsp;
const tagClassesPrimaryCache = weakCache(tagClassesPrimary);
&nbsp;
export function wayPropertiesGen(way: Way, table: EntityTable) {
  const geometry = isArea(way) ? OsmGeometry.AREA : OsmGeometry.LINE;
  const [tagsClass, tagsClassType] = tagClassesPrimaryCache(way.tags);
  const match = presetsMatcherCached(geometry)(way.tags);
  return {
    name: way.tags.get('name') || way.tags.get('ref'),
    icon: match &amp;&amp; match.icon,
    geometry,
    tagsClass,
    tagsClassType
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 29 2017 19:59:11 GMT+0530 (IST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
