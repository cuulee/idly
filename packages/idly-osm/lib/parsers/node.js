"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const getNodeGeometry_1 = require("../helpers/getNodeGeometry");
const presets_1 = require("../presets/presets");
exports.DEFAULT_NODE_ICON = 'circle';
function nodePropertiesGen(node, parentWays) {
    return exports.applyNodeMarkup(getNodeGeometry_1.getNodeGeometry(node.id, parentWays), node.tags);
}
exports.nodePropertiesGen = nodePropertiesGen;
exports.applyNodeMarkup = (geometry, tags) => {
    const match = presets_1.presetsMatcherCached(geometry)(tags);
    return {
        icon: (match && match.icon) || exports.DEFAULT_NODE_ICON,
        name: tags.get('name'),
        geometry
    };
};
//# sourceMappingURL=node.js.map