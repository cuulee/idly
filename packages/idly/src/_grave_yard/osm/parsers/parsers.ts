import { List, Map as ImmutableMap, Set } from 'immutable';

import { Entity } from 'osm/entities/entities';
import { propertiesGen } from 'osm/entities/helpers/properties';
import { tagsFactory } from 'osm/entities/helpers/tags';
import { nodeFactory } from 'osm/entities/node';
import { relationFactory } from 'osm/entities/relation';
import { Way, wayFactory } from 'osm/entities/way';
import { genLngLat } from 'osm/geo_utils/lng_lat';

export type ParentWays = ImmutableMap<string, Set<string>>;

export function calculateParentWays(parentWays: ParentWays, ways: Way[]) {
  console.time('calculateParentWays');
  const x = parentWays.withMutations(p => {
    ways.forEach(w => {
      w.nodes.forEach((n, i) => {
        p.update(n, (s = Set()) => (s = s.add(w.id)));
      });
    });
  });
  console.timeEnd('calculateParentWays');
  return x;
}

export function parseXML(
  xml: XMLDocument,
  parentWays: ParentWays = ImmutableMap()
): {
  entities: Entity[];
  parentWays: ParentWays;
} {
  if (!xml || !xml.childNodes) return;

  const root = xml.childNodes[0];
  const children = root.childNodes;
  const entities: Entity[] = [];
  const nodesXML = [];
  const waysXML = [];
  const relationsXML = [];
  const group = {
    node: [],
    way: [],
    relation: []
  };
  for (let i = 0, l = children.length; i < l; i++) {
    const child = children[i];
    const parser = group[child.nodeName];
    if (parser) group[child.nodeName].push(child);
  }

  group.relation = group.relation.map(parsers.relation);
  group.way = group.way.map(w => parsers.way(w));

  parentWays = calculateParentWays(parentWays, group.way);

  group.node = group.node.map(n => parsers.node(n, parentWays));

  return {
    entities: [...group.node, ...group.way, ...group.relation],
    parentWays
  };
}

function getVisible(attrs) {
  return !attrs.visible || attrs.visible.value !== 'false';
}

function getMembers(obj) {
  const elems = obj.getElementsByTagName('member');
  const members = new Array(elems.length);
  for (let i = 0, l = elems.length; i < l; i++) {
    const attrs = elems[i].attributes;
    members[i] = ImmutableMap({
      id: attrs.type.value[0] + attrs.ref.value,
      type: attrs.type.value,
      role: attrs.role.value
    });
  }
  return List(members);
}

function getNodes(obj) {
  const elems = obj.getElementsByTagName('nd');
  const nodes = new Array(elems.length);
  for (let i = 0, l = elems.length; i < l; i++) {
    nodes[i] = 'n' + elems[i].attributes.ref.value;
  }
  return List(nodes);
}

function getTags(obj) {
  const elems = obj.getElementsByTagName('tag');
  const tags = {};
  for (let i = 0, l = elems.length; i < l; i++) {
    const attrs = elems[i].attributes;
    tags[attrs.k.value] = attrs.v.value;
  }
  return tagsFactory(tags);
}

function getLoc(attrs) {
  const lon = attrs.lon && attrs.lon.value;
  const lat = attrs.lat && attrs.lat.value;
  return genLngLat([parseFloat(lon), parseFloat(lat)]);
}

const parsers = {
  node: function nodeData(obj, parentWays?) {
    const attrs = obj.attributes;
    const id = 'n' + attrs.id.value;
    return nodeFactory({
      id,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        changeset: attrs.changeset && attrs.changeset.value,
        uid: attrs.uid && attrs.uid.value,
        user: attrs.user && attrs.user.value
      }),
      loc: getLoc(attrs),
      tags: getTags(obj)
      // geometry: getNodeGeometry(id, parentWays)
    });
  },

  way: function wayData(obj) {
    const attrs = obj.attributes;
    return wayFactory({
      id: 'w' + attrs.id.value,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        changeset: attrs.changeset && attrs.changeset.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        user: attrs.user && attrs.user.value,
        uid: attrs.uid && attrs.uid.value
      }),
      tags: getTags(obj),
      nodes: getNodes(obj)
    });
    // return way.update('geometry', () => getWayGeometry(way, areaKeys));
  },

  relation: function relationData(obj) {
    const attrs = obj.attributes;
    return relationFactory({
      id: 'r' + attrs.id.value,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        changeset: attrs.changeset && attrs.changeset.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        user: attrs.user && attrs.user.value,
        uid: attrs.uid && attrs.uid.value
      }),
      tags: getTags(obj),
      members: getMembers(obj)
    });
  }
};
