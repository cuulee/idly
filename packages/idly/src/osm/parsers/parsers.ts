import { List, Map } from 'immutable';
import { attachToWindow } from 'utils/attach_to_window';

import { nodeFactory } from 'osm/entities/node';
import { relationFactory } from 'osm/entities/relation';
import { wayFactory } from 'osm/entities/way';
import { genLngLat } from 'osm/geo_utils/lng_lat';
import { propertiesGen } from 'osm/others/properties';
import { tagsFactory } from 'osm/others/tags';

export function parseXML(xml: Document) {
  if (!xml || !xml.childNodes) return;

  const root = xml.childNodes[0];
  const children = root.childNodes;
  const entities = [];
  for (let i = 0, l = children.length; i < l; i++) {
    const child = children[i];
    const parser = parsers[child.nodeName];
    if (parser) {
      entities.push(parser(child));
    }
  }
  return entities;
}

function getVisible(attrs) {
  return !attrs.visible || attrs.visible.value !== 'false';
}

function getMembers(obj) {
  const elems = obj.getElementsByTagName('member');
  const members = new Array(elems.length);
  for (let i = 0, l = elems.length; i < l; i++) {
    const attrs = elems[i].attributes;
    members[i] = Map({
      id: attrs.type.value[0] + attrs.ref.value,
      type: attrs.type.value,
      role: attrs.role.value
    });
  }
  return List(members);
}
function getNodes(obj) {
  const elems = obj.getElementsByTagName('nd');
  const nodes = new Array(elems.length);
  for (let i = 0, l = elems.length; i < l; i++) {
    nodes[i] = 'n' + elems[i].attributes.ref.value;
  }
  return List(nodes);
}

function getTags(obj) {
  const elems = obj.getElementsByTagName('tag');
  const tags = {};
  for (let i = 0, l = elems.length; i < l; i++) {
    const attrs = elems[i].attributes;
    tags[attrs.k.value] = attrs.v.value;
  }
  return tagsFactory(tags);
}

function getLoc(attrs) {
  const lon = attrs.lon && attrs.lon.value;
  const lat = attrs.lat && attrs.lat.value;
  return genLngLat([parseFloat(lon), parseFloat(lat)]);
}

const parsers = {
  node: function nodeData(obj) {
    const attrs = obj.attributes;
    return nodeFactory({
      id: 'n' + attrs.id.value,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        changeset: attrs.changeset && attrs.changeset.value,
        uid: attrs.uid && attrs.uid.value,
        user: attrs.user && attrs.user.value
      }),
      loc: getLoc(attrs),
      tags: getTags(obj)
    });
  },

  way: function wayData(obj) {
    const attrs = obj.attributes;
    return wayFactory({
      id: 'w' + attrs.id.value,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        changeset: attrs.changeset && attrs.changeset.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        user: attrs.user && attrs.user.value,
        uid: attrs.uid && attrs.uid.value
      }),
      tags: getTags(obj),
      nodes: getNodes(obj)
    });
  },

  relation: function relationData(obj) {
    const attrs = obj.attributes;
    return relationFactory({
      id: 'r' + attrs.id.value,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        changeset: attrs.changeset && attrs.changeset.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        user: attrs.user && attrs.user.value,
        uid: attrs.uid && attrs.uid.value
      }),
      tags: getTags(obj),
      members: getMembers(obj)
    });
  }
};
