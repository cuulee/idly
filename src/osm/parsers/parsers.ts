import { List, Map as ImmutableMap } from 'immutable';

import { Geometry } from 'osm/entities/constants';
import { propertiesGen } from 'osm/entities/helpers/properties';
import { tagsFactory } from 'osm/entities/helpers/tags';
import { nodeFactory } from 'osm/entities/node';
import { relationFactory } from 'osm/entities/relation';
import { Way, wayFactory } from 'osm/entities/way';
import { genLngLat } from 'osm/geo_utils/lng_lat';
import { AreaKeys } from 'osm/presets/areaKeys';
// import { calculateParentWays } from 'core/coreOperations';

export type ParentWays = Map<string, Set<string>>;
/**
 * @REVISIT to differentiate between edge vertex and middle vertex
 *  visit here, I guess one should be able to calculate it from this point.
 *  Though I still dont know what would be the best way to figure out
 *  if a node is shared between two ways.
 *  this does not guarantee that the geometry is shared
 */
export function calculateParentWays(parentWays: ParentWays, ways: Way[]) {
  ways.forEach(w => {
    const closed = isClosed(w);
    w.nodes.forEach((n, i) => {
      if (!parentWays.has(n)) parentWays.set(n, new Set([w.id]));
      else {
        parentWays.get(n).add(w.id);
      }
    });
  });
  return parentWays;
}
export function parseXML(
  xml: Document,
  areaKeys: AreaKeys = ImmutableMap(),
  parentWays: ParentWays = new Map()
) {
  if (!xml || !xml.childNodes) return;

  const root = xml.childNodes[0];
  const children = root.childNodes;
  const entities = [];
  const nodesXML = [];
  const waysXML = [];
  const relationsXML = [];
  const group = {
    node: [],
    way: [],
    relation: []
  };
  for (let i = 0, l = children.length; i < l; i++) {
    const child = children[i];
    const parser = group[child.nodeName];
    if (parser) group[child.nodeName].push(child);
  }

  group.relation = group.relation.map(parsers.relation);
  group.way = group.way.map(w => parsers.way(w, areaKeys));

  parentWays = calculateParentWays(parentWays, group.way);

  group.node = group.node.map(n => parsers.node(n, parentWays));

  return {
    entities: [...group.node, ...group.way, ...group.relation],
    parentWays
  };
}

function getVisible(attrs) {
  return !attrs.visible || attrs.visible.value !== 'false';
}

function getMembers(obj) {
  const elems = obj.getElementsByTagName('member');
  const members = new Array(elems.length);
  for (let i = 0, l = elems.length; i < l; i++) {
    const attrs = elems[i].attributes;
    members[i] = ImmutableMap({
      id: attrs.type.value[0] + attrs.ref.value,
      type: attrs.type.value,
      role: attrs.role.value
    });
  }
  return List(members);
}

function getNodes(obj) {
  const elems = obj.getElementsByTagName('nd');
  const nodes = new Array(elems.length);
  for (let i = 0, l = elems.length; i < l; i++) {
    nodes[i] = 'n' + elems[i].attributes.ref.value;
  }
  return List(nodes);
}

function getTags(obj) {
  const elems = obj.getElementsByTagName('tag');
  const tags = {};
  for (let i = 0, l = elems.length; i < l; i++) {
    const attrs = elems[i].attributes;
    tags[attrs.k.value] = attrs.v.value;
  }
  return tagsFactory(tags);
}

function getLoc(attrs) {
  const lon = attrs.lon && attrs.lon.value;
  const lat = attrs.lat && attrs.lat.value;
  return genLngLat([parseFloat(lon), parseFloat(lat)]);
}

const parsers = {
  node: function nodeData(obj, parentWays?) {
    const attrs = obj.attributes;
    const id = 'n' + attrs.id.value;
    return nodeFactory({
      id,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        changeset: attrs.changeset && attrs.changeset.value,
        uid: attrs.uid && attrs.uid.value,
        user: attrs.user && attrs.user.value
      }),
      loc: getLoc(attrs),
      tags: getTags(obj),
      geometry: getNodeGeometry(id, parentWays)
    });
  },

  way: function wayData(obj, areaKeys?) {
    const attrs = obj.attributes;
    const way = wayFactory({
      id: 'w' + attrs.id.value,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        changeset: attrs.changeset && attrs.changeset.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        user: attrs.user && attrs.user.value,
        uid: attrs.uid && attrs.uid.value
      }),
      tags: getTags(obj),
      nodes: getNodes(obj)
    });
    return way.update('geometry', () => getWayGeometry(way, areaKeys));
  },

  relation: function relationData(obj) {
    const attrs = obj.attributes;
    return relationFactory({
      id: 'r' + attrs.id.value,
      properties: propertiesGen({
        visible: getVisible(attrs),
        version: attrs.version.value,
        changeset: attrs.changeset && attrs.changeset.value,
        timestamp: attrs.timestamp && attrs.timestamp.value,
        user: attrs.user && attrs.user.value,
        uid: attrs.uid && attrs.uid.value
      }),
      tags: getTags(obj),
      members: getMembers(obj)
    });
  }
};

/**
 *  @REVISIT
 *   I am not sure about this,
 *   // mosty holds// 1st if <way><n id=X></way> I should have node X in the same request.
 *   // proved wrong// 2nd if node id=Y I should have all the ways having Y in the same request.
 *
 * @REVISIT
 *  need to test / figure out how to handle 2nd point above ^^.
 *  so this guy at sagas would simply filter away this new node information
 *  wrapped in way coming in any subsequent request. hence our node would
 *  not get the new status or VERTEX_SHARED
 */
export function getNodeGeometry(id, parentWays?: ParentWays) {
  if (parentWays && parentWays.get(id))
    return parentWays.get(id).size > 1
      ? Geometry.VERTEX_SHARED
      : Geometry.VERTEX;
  return Geometry.POINT;
}

export function getWayGeometry(way: Way, areaKeys: AreaKeys = ImmutableMap()) {
  if (way.tags.get('area') === 'yes') return Geometry.AREA;

  if (!isClosed(way) || way.tags.get('area') === 'no') return Geometry.LINE;
  const keys = way.tags.keySeq();
  let found = false;
  way.tags.forEach((v, key) => {
    if (areaKeys.has(key) && !areaKeys.get(key).has(v)) {
      found = true;
      return false;
    }
  });
  return found ? Geometry.AREA : Geometry.LINE;
}

export function isClosed(way: Way) {
  return way.nodes.size > 1 && way.nodes.first() === way.nodes.last();
}
