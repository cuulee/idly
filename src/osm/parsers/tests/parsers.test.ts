import { parseXML } from 'osm/parsers/parsers';
import {
  miniWayXML1,
  miniXML1,
  miniXML2,
  miniXML3,
  nodeXML1
} from 'osm/parsers/tests/fixtures';
// const nodeXML =
//   '<?xml version="1.0" encoding="UTF-8"?><osm>' +
//   '<node id="1" version="1" changeset="1" lat="0" lon="0" visible="true" timestamp="2009-03-07T03:26:33Z"></node>' +
//   '</osm>';

const wayXML =
  '<?xml version="1.0" encoding="UTF-8"?><osm>' +
  '<way id="1" visible="true" timestamp="2008-01-03T05:24:43Z" version="1" changeset="522559"><nd ref="1"/></way>' +
  '</osm>';
describe.only('parsers', () => {
  describe('node', () => {
    it('matches snapshot', () => {
      const parser = new DOMParser();
      const xml = parser.parseFromString(miniXML2, 'text/xml');
      expect(parseXML(xml)).toMatchSnapshot();
      expect(
        parseXML(parser.parseFromString(miniXML1, 'text/xml'))
      ).toMatchSnapshot();
    });
    it('matches a bigger example', () => {
      const parser = new DOMParser();
      const XML = nodeXML1;
      const xml = parser.parseFromString(XML, 'text/xml');
      expect(parseXML(xml)).toMatchSnapshot();
    });
  });
  describe('way', () => {
    it('matches snapshot', () => {
      const parser = new DOMParser();
      const xml = parser.parseFromString(miniWayXML1, 'text/xml');
      expect(parseXML(xml)).toMatchSnapshot();
    });
  });
  describe('it makes good parentways', () => {
    it('matches snapshot', () => {
      const parser = new DOMParser();
      const xml = parser.parseFromString(miniXML3, 'text/xml');
      expect(parseXML(xml)).toMatchSnapshot();
    });
  });
});
