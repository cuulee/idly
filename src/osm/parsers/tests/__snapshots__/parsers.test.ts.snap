// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsers node matches a bigger example 1`] = `
Array [
  Object {
    "geometry": "point",
    "id": "n4133782712",
    "loc": Object {
      "lat": 34.5280866,
      "lon": 74.1795357,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782713",
    "loc": Object {
      "lat": 34.528206,
      "lon": 74.1807924,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782714",
    "loc": Object {
      "lat": 34.5283653,
      "lon": 74.1822907,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782715",
    "loc": Object {
      "lat": 34.528425,
      "lon": 74.1815174,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782716",
    "loc": Object {
      "lat": 34.5285445,
      "lon": 74.1788832,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782717",
    "loc": Object {
      "lat": 34.5290024,
      "lon": 74.1788349,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782718",
    "loc": Object {
      "lat": 34.5292015,
      "lon": 74.179584,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782719",
    "loc": Object {
      "lat": 34.529301,
      "lon": 74.1824357,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782720",
    "loc": Object {
      "lat": 34.5294603,
      "lon": 74.1794149,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782721",
    "loc": Object {
      "lat": 34.5296992,
      "lon": 74.1823873,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782722",
    "loc": Object {
      "lat": 34.5297191,
      "lon": 74.1788349,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782723",
    "loc": Object {
      "lat": 34.5300377,
      "lon": 74.1821698,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782724",
    "loc": Object {
      "lat": 34.5303562,
      "lon": 74.1783274,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782725",
    "loc": Object {
      "lat": 34.530396,
      "lon": 74.1816382,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782726",
    "loc": Object {
      "lat": 34.5305354,
      "lon": 74.1805024,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782727",
    "loc": Object {
      "lat": 34.5309137,
      "lon": 74.1799224,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782728",
    "loc": Object {
      "lat": 34.5311526,
      "lon": 74.1786899,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782729",
    "loc": Object {
      "lat": 34.5314114,
      "lon": 74.1793182,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:40Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "line",
    "id": "w411770875",
    "nodes": Array [
      "n4133782719",
      "n4133782714",
      "n4133782715",
      "n4133782713",
      "n4133782712",
      "n4133782716",
      "n4133782717",
      "n4133782718",
      "n4133782720",
      "n4133782722",
      "n4133782724",
      "n4133782728",
      "n4133782729",
      "n4133782727",
      "n4133782726",
      "n4133782725",
      "n4133782723",
      "n4133782721",
      "n4133782719",
    ],
    "properties": Object {
      "changeset": "50089927",
      "timestamp": "2017-07-06T17:19:04Z",
      "uid": "2815653",
      "user": "Srihari Thalla",
      "version": "2",
      "visible": true,
    },
    "tags": Object {
      "landuse": "residential",
      "place": "village",
      "source": "MSF Survey",
    },
    "type": "way",
  },
]
`;

exports[`parsers node matches snapshot 1`] = `
Array [
  Object {
    "geometry": "point",
    "id": "n1",
    "loc": Object {
      "lat": 0,
      "lon": 0,
    },
    "properties": Object {
      "changeset": "1",
      "timestamp": "2009-03-07T03:26:33Z",
      "uid": undefined,
      "user": undefined,
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
]
`;

exports[`parsers node matches snapshot 2`] = `
Array [
  Object {
    "geometry": "point",
    "id": "n4133782712",
    "loc": Object {
      "lat": 34.5280866,
      "lon": 74.1795357,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
  Object {
    "geometry": "point",
    "id": "n4133782713",
    "loc": Object {
      "lat": 34.528206,
      "lon": 74.1807924,
    },
    "properties": Object {
      "changeset": "38701677",
      "timestamp": "2016-04-19T17:58:39Z",
      "uid": "592097",
      "user": "Dombo",
      "version": "1",
      "visible": true,
    },
    "tags": Object {},
    "type": "node",
  },
]
`;

exports[`parsers way matches snapshot 1`] = `
Array [
  Object {
    "geometry": "line",
    "id": "w411770875",
    "nodes": Array [
      "n4133782719",
      "n4133782714",
      "n4133782715",
      "n4133782713",
      "n4133782712",
      "n4133782716",
      "n4133782717",
      "n4133782718",
      "n4133782720",
      "n4133782722",
      "n4133782724",
      "n4133782728",
      "n4133782729",
      "n4133782727",
      "n4133782726",
      "n4133782725",
      "n4133782723",
      "n4133782721",
      "n4133782719",
    ],
    "properties": Object {
      "changeset": "50089927",
      "timestamp": "2017-07-06T17:19:04Z",
      "uid": "2815653",
      "user": "Srihari Thalla",
      "version": "2",
      "visible": true,
    },
    "tags": Object {
      "landuse": "residential",
      "place": "village",
      "source": "MSF Survey",
    },
    "type": "way",
  },
]
`;
